src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,127 @@
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validLogin");
+        firstUser.setPassword("validPassword");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUserWithSameLogin = new User();
+        secondUserWithSameLogin.setLogin("validLogin");
+        secondUserWithSameLogin.setPassword("anotherPassword");
+        secondUserWithSameLogin.setAge(25);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(secondUserWithSameLogin);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userWithShortPassword = new User();
+        userWithShortPassword.setLogin("validLogin");
+        userWithShortPassword.setPassword("pwd");
+        userWithShortPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        });
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        User userWithNullPassword = new User();
+        userWithNullPassword.setLogin("validLogin");
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        User userWithNullLogin = new User();
+        userWithNullLogin.setLogin(null);
+        userWithNullLogin.setPassword("validPassword");
+        userWithNullLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullLogin);
+        });
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        User userWithNullAge = new User();
+        userWithNullAge.setLogin("validLogin");
+        userWithNullAge.setPassword("validPassword");
+        userWithNullAge.setAge(0);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullAge);
+        });
+    }
+
+    @Test
+    void register_fewYears_notOk() {
+        User userWithFewYears = new User();
+        userWithFewYears.setLogin("validLogin");
+        userWithFewYears.setPassword("validPassword");
+        userWithFewYears.setAge(16);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithFewYears);
+        });
+    }
```

Comment:
You are testing the case where the age is less than the minimum age, but you are not testing the edge case where the age is exactly the minimum age. It would be good to add a test for this case to ensure that the age check is working correctly.